inc/azure_c_shared_utility/tlsio_schannel.h

```diff
@@ -14,15 +14,27 @@ extern "C" {
 #include <stddef.h>
 #endif /* __cplusplus */
 
-MOCKABLE_FUNCTION(, CONCRETE_IO_HANDLE, tlsio_schannel_create, void*, io_create_parameters);
-MOCKABLE_FUNCTION(, void, tlsio_schannel_destroy, CONCRETE_IO_HANDLE, tls_io);
-MOCKABLE_FUNCTION(, int, tlsio_schannel_open, CONCRETE_IO_HANDLE, tls_io, ON_IO_OPEN_COMPLETE, on_io_open_complete, void*, on_io_open_complete_context, ON_BYTES_RECEIVED, on_bytes_received, void*, on_bytes_received_context, ON_IO_ERROR, on_io_error, void*, on_io_error_context);
-MOCKABLE_FUNCTION(, int, tlsio_schannel_close, CONCRETE_IO_HANDLE, tls_io, ON_IO_CLOSE_COMPLETE, on_io_close_complete, void*, callback_context);
-MOCKABLE_FUNCTION(, int, tlsio_schannel_send, CONCRETE_IO_HANDLE, tls_io, const void*, buffer, size_t, size, ON_SEND_COMPLETE, on_send_complete, void*, callback_context);
-MOCKABLE_FUNCTION(, void, tlsio_schannel_dowork, CONCRETE_IO_HANDLE, tls_io);
-MOCKABLE_FUNCTION(, int, tlsio_schannel_setoption, CONCRETE_IO_HANDLE, tls_io, const char*, optionName, const void*, value);
-
-MOCKABLE_FUNCTION(, const IO_INTERFACE_DESCRIPTION*, tlsio_schannel_get_interface_description);
+#define TLSIO_STATE_VALUES                        \
+    TLSIO_STATE_NOT_OPEN,                         \
+    TLSIO_STATE_OPENING_UNDERLYING_IO,            \
+    TLSIO_STATE_HANDSHAKE_CLIENT_HELLO_SENT,      \
+    TLSIO_STATE_HANDSHAKE_SERVER_HELLO_RECEIVED,  \
+    TLSIO_STATE_RENEGOTIATE,                      \
+    TLSIO_STATE_OPEN,                             \
+    TLSIO_STATE_CLOSING,                          \
+    TLSIO_STATE_ERROR
+
+DEFINE_ENUM(TLSIO_STATE, TLSIO_STATE_VALUES);
```

Comment:
Sneaky. This change has nothing to do with DLLEXPORT.
And it should not go in as the states are actually not used in the interface anywhere. Thus they are not really mandated.

Reply:
Actually not so sneaky. In order export TLSIO_STATE_FromString and TLSIO_STATEStrings I had to put the define in the header file since the export macro is only used in DEFINE_ENUM. All of these enum string functions are exported for the various return values. If they are not used in the interface anywhere then I think it should have been defined with DEFINE_LOCAL_ENUM but I'm not sure. Those are not exported.

---
In reply to: [184158329](https://github.com/Azure/azure-c-shared-utility/pull/152#discussion_r184158329) [](ancestors = 184158329)

Reply:
What I mean is: since the DEFINE_ENUM was not in there up to now,m why does it need to be in now?

---
In reply to: [184223100](https://github.com/Azure/azure-c-shared-utility/pull/152#discussion_r184223100) [](ancestors = 184223100,184218857,184158329)